src/codewhisperer/tracker/codewhispererCodeCoverageTracker.ts

```diff
@@ -238,21 +240,63 @@ export class CodeWhispererCodeCoverageTracker {
         this.addTotalTokens(filename, text.length)
     }
 
+    // For below 2 edge cases
+    // 1. newline character with indentation
+    // 2. 2 character insertion of closing brackets
+    public getCharacterCountFromComplexEvent(e: vscode.TextDocumentChangeEvent) {
+        if (e.contentChanges.length === 2) {
+            const text1 = e.contentChanges[0].text
+            const text2 = e.contentChanges[1].text
+            if (text1.length === 0) {
+                if (text2.startsWith('\n') && text2.trim().length === 0) {
+                    return 1
+                }
+                if (autoClosingKeystrokeInputs.includes(text2)) {
+                    return 2
+                }
+            }
+            if (text2.length === 0) {
```

Comment:
if `text1 !== 0 && text2 !== 0` , `getCharacterCountFromComplexEvent` will return `0`. is that expected?

Reply:
Yes. When user input one key stroke, either text1 is empty or text2 is empty.

Reply:
but i'm asking about the case where both are _not_ empty

Reply:
>where both are not empty

In this case, it won't be a user keystroke input. I have never seen this case btw.